src/coreComponents/mesh/Perforation.cpp

```diff
@@ -46,6 +46,11 @@ Perforation::Perforation( string const & name, Group * const parent )
     setApplyDefaultValue( 0.0 ).
     setInputFlag( InputFlags::OPTIONAL ).
     setDescription( "Perforation skin factor" );
+
+  registerWrapper( viewKeyStruct::targetRegionString(), &m_targetRegionName ).
+    setRTTypeName( rtTypes::CustomTypes::groupNameRef ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Target region to connect the perforation" );
```

Comment:
Why can it be customized by the user? Cannot it be determined automatically?

Reply:
the idea here is to be able to connect well to a fracture
fractures are geometrically zero volume objects so a perforation that lies on a fracture may numerically also get connected to a cell block next to it
to avoid this situation and to allow the user to control the behavior, `targetRegion` can be optionally specified
when not provided, GEOS will try to connect perforation with all the regions